### ISSUE TITLE
Indentation error in SASS maps

### ISSUE THREAD

**Issue by cyril-lamotte** [stylelint/stylelint#1696]
Hi, thank you for your work.

Is-it a regression ?

Since some days, that notation is giving indentations errors (SASS) : 

```scss
// SASS map
$breakpoints: (
  '0' : 0,
  'xs': 480,
  'sm': 768,
  'md': 992,
  'lg': $max-site-width-int
);
```

Is-it a regression ?

---

Version of stylelint : 7.0.3 with gulp-stylelint

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS maps.

> What did you expect to happen?

No warnings to be flagged

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
 22:3  ×  Expected indentation of 0 spaces   indentation
 23:3  ×  Expected indentation of 0 spaces   indentation
 24:3  ×  Expected indentation of 0 spaces   indentation
 25:3  ×  Expected indentation of 0 spaces   indentation
 26:3  ×  Expected indentation of 0 spaces   indentation
```

**Comment by jeddy3** [comment]
> What stylelint configuration is needed to reproduce this issue?

**Comment by cyril-lamotte** [comment]
Hi,

```{
  'rules': {
    'block-no-empty': true,
    'block-opening-brace-newline-after': 'always-multi-line',

    'color-no-invalid-hex': true,
    'color-hex-case': 'lower',

    'number-leading-zero': 'always',
    'number-no-trailing-zeros': true,
    'length-zero-no-unit': true,

    'declaration-bang-space-after': 'never',
    'declaration-bang-space-before': 'always',
    'declaration-colon-newline-after': 'always-multi-line',
    'declaration-colon-space-before': 'never',
    'declaration-colon-space-after': 'always',
    'declaration-block-no-duplicate-properties': [true, {
      "ignore" : ["consecutive-duplicates"]
    }],
    'declaration-block-semicolon-space-before': 'never',
    'declaration-block-semicolon-newline-after': 'always-multi-line',
    'declaration-block-semicolon-newline-before': 'never-multi-line',
    'declaration-block-trailing-semicolon': 'always',

    'block-closing-brace-newline-after': 'always',
    'block-closing-brace-newline-before': 'always-multi-line',
    'block-opening-brace-space-before': 'always-multi-line',
    'block-opening-brace-space-after': 'always-single-line',

    'selector-attribute-brackets-space-inside': 'never',
    'selector-attribute-operator-space-before': 'never',
    'selector-attribute-operator-space-after': 'never',
    'selector-combinator-space-after': 'always',
    'selector-combinator-space-before': 'always',
    'selector-max-empty-lines': 0,
    'selector-pseudo-element-case': 'lower',
    'selector-list-comma-newline-before': 'never-multi-line',
    'selector-list-comma-newline-after': 'always',
    'selector-list-comma-space-before': 'never',

    'media-feature-colon-space-after': 'always',
    'media-feature-colon-space-before': 'never',
    'media-feature-parentheses-space-inside': 'never',

    'no-extra-semicolons': true,

    'indentation': [2],
    'max-empty-lines': 5,
    'unit-whitelist': ['em', 'rem', '%', 'px', 's', 'ms', 'deg'],

    'value-no-vendor-prefix': true,
    'value-keyword-case': 'lower',
    'value-list-comma-space-after': 'always'
  }
}
```

**Comment by jeddy3** [comment]
Try:

`'indentation': 2,`

It's a primary option by itself and shouldn't be wrapped in an array.

**Comment by cyril-lamotte** [comment]
Tried it, but it is not working.
The fact is the the same config was working 2 weeks ago. I just updated to version 7 that's why I think there is a regression.

**Comment by evilebottnawi** [comment]
@cyril-lamotte latest stylelint with `'indentation': 4` and 
```scss
$breakpoints: (
    "0" : 0,
    "xs": 480,
    "sm": 768,
    "md": 992,
    "lg": $max-site-width-int
);
```
Works fine, can't reproduce, can you create test repo with reproduced error?

**Comment by jeddy3** [comment]
> I just updated to version 7 that's why I think there is a regression.

The behaviour of the `indentation` rule [was changed](http://stylelint.io/CHANGELOG/#700) in `7.0.0`. It now checks the indentation inside of parentheses by default.

The behaviour of the primary options was also tweaked in `7.0.0`, to [allow plugins](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md#allow-primary-option-arrays) to use arrays for their primary options. Which is why I wanted to rule out that it wasn't a configuration issue first :)

As @evilebottnawi wasn't able to reproduce it for that code snippet and that we also have [tests](https://github.com/stylelint/stylelint/blob/master/src/rules/indentation/__tests__/functions.js#L36-L42) for it.  All I can think of is perhaps there is some SCSS code before that snippet in your file that is causes the errors within the map.

If you can create a reduced-test repo with a reproduced error for us, then we can try to get to the bottom of this for you :)

**Comment by cyril-lamotte** [comment]
Hi,

[the test Repo](https://github.com/cyril-lamotte/stylelint-bug)

Thanks for your help

**Comment by jeddy3** [comment]
@cyril-lamotte Thanks for producing the test repo!

Umm, no dice I'm afraid though. I was able to run `gulp lint-css` *without errors*. 

I tweaked `style.scss` to:

```scss
$breakpoints: (
  '0' : 0,
'xs': 480,
  'sm': 768,
  'md': 992
);
```

And correctly got:

```console
sources/scss/style.scss
 3:1  ✖  Expected indentation of 2 spaces   indentation
```

So, I know the tools in your test repo are working correctly.

I tried on node `v6.3.0` and `v5.9.0`.

`npm ls --depth=1` gives me:

```console
├─┬ gulp-stylelint@3.0.0
│ ├── deep-extend@0.4.1
│ ├── mkdirp@0.5.1
│ ├── promise@7.1.1
│ └── stylelint@7.0.3
```

Can you see anything above that's different from your machine that might account for the why I can running the test repo without issue, but you can't? I'm on OSX, what about you?

I'm clutching at straws here, but can you delete your `node_modules` folder and also *uninstall* any global copies of stylelint (e.g. `npm uninstall -g stylelint`) you might have, and then try running the test repo again?

**Comment by cyril-lamotte** [comment]
I'm on windows 7, with node 5.5.0/, these are the only differences. I'm going to update node.

**Comment by cyril-lamotte** [comment]
After updating, deleting node_module & doing the cache clean the problem is still here !
I do not understand but I'll try with an another environnement at home.

**Comment by evilebottnawi** [comment]
@cyril-lamotte you can testing with https://github.com/coreybutler/nvm-windows, it allow fast change node and npm version, also see where modules loaded.

**Comment by jeddy3** [comment]
> I do not understand but I'll try with an another environnement at home.

Good plan. Please report back and we'll go from there.

**Comment by cyril-lamotte** [comment]
Hi,

I confirm that it's working on an other environment (macOS X).
So maybe it's windows...

**Comment by jeddy3** [comment]
Consolidating this discussion into https://github.com/stylelint/stylelint/issues/1708 as it appears to be the same issue. 

@cyril-lamotte Help resolving this problem in that issue would be greatly appreciated. Our next step is to rule out the node version being a factor. Can you update your version of node to `6` and report back in the other issue please?
